SeriesCruncher/app/src/main/java/com/balazs_csernai/seriescruncher/seriesdetails/ui/EpisodeAdapter.java

```diff
@@ -75,27 +74,30 @@ public void onBindViewHolder(SeasonsViewHolder holder, int position) {
             ViewUtils.visible(holder.airDate);
             holder.airDate.setText(DateUtils.parseDate(item.getAirDate()));
         }
-        holder.itemView.setTag(position);
+        holder.itemView.setTag(mapVisiblePositionToInnerPosition(position));
         holder.itemView.setOnClickListener(this);
     }
 
     @Override
     public int getItemCount() {
-        return episodes.getItemCount();
+        int visibleItemCount = 0;
+        for (EpisodeListItemModel episode : items) {
+            if (isVisible(episode)) {
+                visibleItemCount++;
+            }
+        }
+        return visibleItemCount;
     }
 
     @Override
     public void onClick(View view) {
         int position = (int) view.getTag();
-        episodes.toggle(position);
-        notifyDataSetChanged();
+        onItemTap(position);
     }
 
     public class SeasonsViewHolder extends RecyclerView.ViewHolder {
```

Comment:
Please consider using package private visibility modifier and a static modifier.